src/net/tls.cc

```diff
@@ -1217,30 +1217,27 @@ class session : public enable_lw_shared_from_this<session> {
         });
     }
 
-    future<> do_handshake() {
-        if (_connected) {
-            return make_ready_future<>();
-        }
+    future<> do_handshake2(int (*func)(gnutls_session_t)) {
```

Comment:
```suggestion
    future<> do_handshake(int (*func)(gnutls_session_t)) {
```
Why there is a need to introduce `2`?

Reply:
Here the point is they are privates; Perhaps "do_handshake_invoke" would be better?